### PR TITLE
Bug back command

### LINKED ISSUE #78: Wrong Back Command when module and chapter have same name
The back command jumps from chapter level directly to the admin level if module and chapter name are the same.

### PR DESCRIPTION
**Pull request by gua-guargia**
fixes #78

### CODE REVIEW

File: src/main/java/access/Access.java [hunk 1]
```diff
@@ -110,65 +106,38 @@ public void setIsChapterLevel() {
...
-
         if (isAdminLevel) {
-            if (moduleLevel.equals("")) {
-                throw new IncorrectAccessLevelException("Sorry, you are already in the admin level.");
-            }
             this.moduleLevel = moduleLevel;
             this.level = level + "/" + moduleLevel;
             this.module = new Module(moduleLevel);
             this.isModuleLevel = true;
             this.isAdminLevel = false;
```
Comment (Jane-Ng) [inline src/main/java/access/Access.java]: you can use setIsModuleLevel() to do this

File: src/main/java/access/Access.java [hunk 2]
```diff
@@ -110,65 +106,38 @@ public void setIsChapterLevel() {
...
+            return;
         }
+
+        String replacement = "/" + this.moduleLevel;
+        this.level = level.replace(replacement, "");
+        this.moduleLevel = moduleLevel;
+        this.module = null;
+        this.isModuleLevel = false;
+        this.isAdminLevel = true;
```
Comment (Jane-Ng) [inline src/main/java/access/Access.java]: you can use setIsAdminLevel() to do this

**Review by woozije — COMMENTED** [review]
Looks good to merge

**Review by Jane-Ng — COMMENTED** [review]
Just a suggestion, would it be better for setModuleLevel to be setGoModuleLevel when you go to module and setBackAdminLevel when you back from module level to admin level?

Or if you think setModuleLevel is fine on its own, maybe can have 2 if clauses. something like
if (isModuleLevel) {
    //code
}

if( (isAdminLevel) {
    //code
}

same goes for setChapterLevel

File: src/main/java/access/Access.java [hunk 3]
```diff
@@ -110,65 +106,38 @@ public void setIsChapterLevel() {
...
-                throw new IncorrectAccessLevelException("Sorry, you are already in the chapter level, "
-                        + "please go back to module level first.");
-            }
-            String replacement = "/" + this.chapterLevel;
-            this.level = level.replace(replacement, "");
+            this.level = adminLevel + "/" + moduleLevel;
             this.chapterLevel = chapterLevel;
             this.chapter = null;
             this.isChapterLevel = false;
             this.isModuleLevel = true;
```
Comment (Jane-Ng) [inline src/main/java/access/Access.java]: you can use setIsModuleLevel() to do this

File: src/main/java/access/Access.java [hunk 4]
```diff
@@ -110,65 +106,38 @@ public void setIsChapterLevel() {
...
-            this.level = level + "/" + chapterLevel;
-            this.chapter = new Chapter(chapterLevel);
-            this.isChapterLevel = true;
-            this.isModuleLevel = false;
-        }
+        this.chapterLevel = chapterLevel;
+        this.level = level + "/" + chapterLevel;
+        this.chapter = new Chapter(chapterLevel);
+        this.isChapterLevel = true;
+        this.isModuleLevel = false;
```
Comment (Jane-Ng) [inline src/main/java/access/Access.java]: you can use setIsChaptereLevel() to do this

File: src/main/java/access/Access.java (hunk 1 above)
Comment (gua-guargia) [reply src/main/java/access/Access.java]: Thank you, I have changed all the separated code into setIsModuleLevel().

**Pull request merged**